hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -63,34 +76,117 @@ public UniqueTokenStore(final EntityIdSource ids,
 	) {
 		super(ids, validator, properties, tokens, tokenRelsLedger);
 		this.uniqueTokenSupplier = uniqueTokenSupplier;
+		this.provisionalUniqueTokens = new ArrayList<>(); // TODO MAX_SIZE
+		this.lastMintedSerialNumbers = new ArrayList<>();
+	}
+
+	@Override
+	public List<Long> popLastMintedSerialNumbers() {
+		var copy = new ArrayList<>(lastMintedSerialNumbers);
+		lastMintedSerialNumbers.clear();
+		return copy;
+	}
+
+	@Override
+	public ResponseCodeEnum wipe(final AccountID aId, final TokenID tId, final long wipingAmount, final boolean skipKeyCheck) {
+		return null;
+	}
+
+	@Override
+	public ResponseCodeEnum mintProvisional(final TokenMintTransactionBody txBody, final RichInstant creationTime) {
+		return tokenSanityCheck(txBody.getToken(), merkleToken -> {
+			if (!merkleToken.hasSupplyKey()) {
+				return TOKEN_HAS_NO_SUPPLY_KEY;
+			}
+			final var eId = EntityId.fromGrpcTokenId(txBody.getToken());
+			final var owner = merkleToken.treasury();
+			final var metadataList = txBody.getMetadataList();
+			metadataList.stream().map(el -> {
+				String metaAsStr = el.toStringUtf8();
+				final long serialNum = merkleToken.getCurrentSerialNum(); // create them without incrementing serialNum
```

Comment:
serial number would need to be incremented for each NFT, but should not be incremented on MerkleToken unless we do `commit`